frontend/main.js

```diff
@@ -103,19 +106,60 @@ async function startApp () {
     // Since a runtime error just occured, we likely want to persist app logs to local storage now.
     sbp('appLogs/save')
   }
+
+  // Set up event listeners to keep local (Vuex) and Chelonia states in sync
+  sbp('chelonia/externalStateSetup', { reactiveSet: Vue.set, reactiveDel: Vue.delete })
+
+  // Load Chelonia state (this needs to be done in the SW when Chelonia is
+  // running there)
+  // We only load Chelonia state when SETTING_CURRENT_USER is set because,
+  // currently, Chelonia only has meaningful persistent state when there's
+  // an active session. If there's no logged in user, it means that there's
+  // no state to restore.
+  await sbp('gi.db/settings/load', 'CHELONIA_STATE').then(async (cheloniaState) => {
+    if (!cheloniaState) return
+    const identityContractID = await sbp('gi.db/settings/load', SETTING_CURRENT_USER)
+    if (!identityContractID) return
```

Comment:
Sorry, I still don't get this... What about all of the code in `gi.app/identity/login` (and related selectors and events)? Isn't that supposed to be loading the Chelonia state? What then is this code all about?

Could you please explain in a more elaborate comment?

Reply:
What you are referring to is loading Chelonia state during _login_. This is loading Chelonia state when there already is an active session.

Reply:
But `gi.app/identity/login` is called on every page load, including when there is an active session, is it not?

Reply:
Yes, but it doesn't load Chelonia state in that case

Reply:
Expanded comment

Reply:
Can this code be moved into `controller/app/identity.js` then?

Reply:
No, it cannot. See where the comment starts, it says this needs to be in the service worker. Chelonia state is part of the service worker and there is a single one. This state cannot be set every time there is a page refresh.

Reply:
OK, on a somewhat related note, I'm starting to think that this logic is important and common enough that it would need to be done by every Chelonia-based app, right?

I've created an issue related to this: https://github.com/okTurtles/group-income/issues/2162

Maybe we can get this done for this PR too.

Reply:
It is common logic in the sense that restoring the state may be needed, but the specifics of how it's done depend on the application. For example, here it's checking `SETTING_CURRENT_USER` which would not always exist.

Also, I don't see the connection between this and #2162. Using `chelonia.db` or `chelonia/db` would not address this because it's the state for Chelonia itself that's being loaded. In the client, when the lightweight client option is enabled, `chelonia.db` is mostly a no-op, so it wouldn't work here.